front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -597,49 +592,31 @@ function AssitantDetailViewLink(assistant: LightAgentConfigurationType) {
   );
 }
 
-function Citations({
+function getCitations({
   activeReferences,
   lastHoveredReference,
 }: {
   activeReferences: {
     index: number;
-    document: RetrievalDocumentType | WebsearchResultType;
+    document: MarkdownCitation;
   }[];
   lastHoveredReference: number | null;
 }) {
   activeReferences.sort((a, b) => a.index - b.index);
-  return (
-    <div
-      className="grid grid-cols-3 items-stretch gap-2 pb-4 pt-8 md:grid-cols-4"
-      // ref={citationContainer}
-    >
-      {activeReferences.map(({ document, index }) => {
-        const [documentCitation] =
-          "link" in document
-            ? [
-                {
-                  href: document.link,
-                  title: document.title,
-                  type: "document" as const,
-                },
-              ]
-            : makeDocumentCitations([document]);
-
-        return (
-          <Citation
-            key={index}
-            size="xs"
-            sizing="fluid"
-            isBlinking={lastHoveredReference === index}
-            type={documentCitation.type}
-            title={documentCitation.title}
-            href={documentCitation.href}
-            index={index}
-          />
-        );
-      })}
-    </div>
-  );
+  return activeReferences.map(({ document, index }) => {
```

Comment:
Expected that we are changing the grid layout here?

Reply:
We're using the grid from ConversationMessageContent to get the same layout. Indeed we have here 3 cols and ConversationMessageContent defines a 4 cols grid .. checking

Reply:
We still have a 4 cols grid, layout is almost identical, just a padding slightly different - but it was inconsistent with user content fragment before, now they look aligned (using the same code) :

before : 
![Screenshot 2024-10-25 at 16 24 35](https://github.com/user-attachments/assets/478a5c12-ed6d-4473-b36c-5d3ad57aa93d)

after : 
![Screenshot 2024-10-25 at 16 24 19](https://github.com/user-attachments/assets/b47fcbef-668f-49e1-bd6a-60f29a78c8db)

Reply:
https://dust.tt/w/0ec9852c2f/assistant/WI5uR9uXoI and https://front-edge.dust.tt/w/0ec9852c2f/assistant/WI5uR9uXoI